internal/examples/agent/agent/agent.go

```diff
@@ -354,3 +394,60 @@ func (agent *Agent) onMessage(ctx context.Context, msg *types.MessageData) {
 		}
 	}
 }
+
+func (agent *Agent) onOpampConnectionSettings(ctx context.Context, settings *protobufs.OpAMPConnectionSettings) error {
+	if settings == nil || settings.Certificate == nil {
+		agent.logger.Debugf("Received nil certificate offer, ignoring.\n")
+	}
+
+	cert, err := agent.getCertFromSettings(settings.Certificate)
+	if err != nil {
+		return err
+	}
+
+	agent.logger.Debugf("Reconnecting to verify offered client certificate.\n")
+
+	agent.disconnect()
+	agent.opampClientCert = cert
+	if err := agent.connect(); err != nil {
+		agent.logger.Errorf("Cannot connect using offered certificate: %s. Ignoring the offer\n", err)
+		agent.opampClientCert = nil
```

Comment:
Adding a TODO to take a copy of the settings might be a good idea and restore here.

Reply:
What exactly do you want to prevent by that? Are you worried about the new server pushing connection settings too?

Reply:
To not lose the old connection certs, if any. Say the server offered certificates two times with different settings, the first one was successful, and when the second offer attempt fails, do we go back to the last step, which had working settings or reset to nil? I think it is fine for this example.

Reply:
Can be addressed in issue https://github.com/open-telemetry/opamp-go/issues/178